src/vm/typehandle.cpp

```diff
@@ -702,39 +704,66 @@ BOOL TypeHandle::CanCastTo(TypeHandle type, TypeHandlePairList *pVisited)  const
     {
         THROWS;
         GC_TRIGGERS;
+        MODE_ANY;
         INJECT_FAULT(COMPlusThrowOM());
 
         LOADS_TYPE(CLASS_DEPENDENCIES_LOADED);
     }
     CONTRACTL_END
 
     if (*this == type)
-        return(true);
+        return true;
 
-    if (IsTypeDesc())
-        return AsTypeDesc()->CanCastTo(type, pVisited);
-                
-    if (type.IsTypeDesc())
-        return(false);
+    bool isTypeDesc = IsTypeDesc();
+    if (!isTypeDesc && type.IsTypeDesc())
+        return false;
+        
+    {
+        GCX_COOP();
```

Comment:
I have some concerns about COOP switch here. 
It is a redundancy for coop callers, which are the majority. However, most callers that seem perf sensitive, perform a separate cache lookup first, so this should be rare/slow path for them.
I am hoping this is not too expensive. Otherwise we may want a separate entry point for the few callers that are ANY.